src/command_modules/azure-cli-resource/azure/cli/command_modules/resource/_validators.py

```diff
@@ -36,11 +36,21 @@ def process_deployment_create_namespace(namespace):
     _validate_deployment_name(namespace)
 
 
-def internal_validate_lock_parameters(resource_group, resource_provider_namespace,
+def internal_validate_lock_parameters(namespace, resource_group, resource_provider_namespace,
                                       parent_resource_path, resource_type, resource_name):
     if resource_group is None:
         if resource_name is not None:
-            raise CLIError('--resource-name is ignored if --resource-group is not given.')
+            from msrestazure.tools import parse_resource_id, is_valid_resource_id
+            if not is_valid_resource_id(resource_name):
+                raise CLIError('--resource is not a valid resource ID. '
+                               '--resource as a resource name is ignored if --resource-group is not given.')
+            # resource-name is an ID, populate namespace
+            id_dict = parse_resource_id(resource_name)
+            for id_part in ['resource_name', 'resource_type', 'resource_group']:
+                setattr(namespace, id_part, id_dict.get(id_part))
+            setattr(namespace, 'resource_provider_namespace', id_dict.get('resource_namespace'))
+            setattr(namespace, 'parent_resource_path', id_dict.get('resource_parent').strip('/'))
```

Comment:
This would remove both the start and end `/`.
Is there a case where the leading / should not be removed?
Consider rstrip?
https://docs.python.org/2/library/string.html#string.rstrip